src/engine/loopingcontrol.cpp

```diff
@@ -835,27 +923,104 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint) {
         } else {
             newloopSamples.end += 2;
         }
-    } else if (newloopSamples.end > samples) {
+    }
+    if (newloopSamples.end > samples) {
         // Do not allow beat loops to go beyond the end of the track
         newloopSamples.end = samples;
```

Comment:
This looks wrong. Is there any use-case for this? 

The user has probability accidentally doubled the loop size beyond the track end. 
If we now cut the loop to a not beat matching value, the track will fall out of sync. 
I think a better strategy is to not allow to make the loop that wide and keep the track in sync.